src/messaging/ReliableMessageMgr.cpp

```diff
@@ -520,6 +559,16 @@ void ReliableMessageMgr::CalculateNextRetransTime(RetransTableEntry & entry)
                     backoff.count(), peerIsActive ? "Active" : "Idle", config.mIdleRetransTimeout.count(),
                     config.mActiveRetransTimeout.count(), config.mActiveThresholdTime.count());
 #endif // CHIP_PROGRESS_LOGGING
+
+#if CHIP_CONFIG_MRP_ANALYTICS_ENABLED
+    // For initial send the packet has already been submitted to transport layer successfully.
+    // On re-transmits we do not know if transport layer is unable to retransmit for some
+    // reason, so saying we have sent re-transmit here is a little presumptuous.
+    ReliableMessageAnalyticsDelegate::EventType eventType = entry.sendCount == 0
+        ? ReliableMessageAnalyticsDelegate::EventType::kInitialSend
+        : ReliableMessageAnalyticsDelegate::EventType::kRetransmission;
```

Comment:
would it be helpful to send how many retransmissions? Imagining statistics of "1 retransmission" vs "2 retransmissions" and such. Maybe we can see "if the 2nd retransmission happens, how often do they really succeeed vs totally failing?"

Reply:
Do you mind if that is done in a follow up? I want to keep this PR to a minimal and Delegate parameter `TransmitEvent` was implemented in a way to make adding additional analytics later on is simple

Reply:
I will leave this out in this PR